frontend/src/utils/CustomerUtils.ts

```diff
@@ -25,3 +25,18 @@ export const randomColor = (
 export const getCheckedIds = <T>(array: { checked: boolean; id: T }[]) => {
   return array.filter(({ checked }) => checked).map(({ id }) => id);
 };
+
+export const targetCustomerStakes = (customers: Customer[]) => {
+  const totalValue = customers.reduce((acc, { weight }) => acc + weight, 0);
+  return [...customers]
+    .sort((a, b) => b.weight - a.weight)
+    .map(({ id, weight }) => ({
+      id,
+      percent: weight / totalValue,
+    }));
+};
+
+export const stakesDifferTooMuchFromPlanned = (
+  decimal: number,
+  targetDecimal: number,
```

Comment:
The parameter names could be a bit more descriptive e.g.
```suggestion
  actual: number,
  planned: number,
```